controllers/drplacementcontrol_controller.go

```diff
@@ -606,22 +606,36 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, errorswrapper.Wrap(err, "failed to get DRPC object")
 	}
 
+	drPolicy, err := r.getDRPolicy(ctx, drpc, logger)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get DRPolicy %w", err)
+	}
+
+	if !drPolicy.ObjectMeta.DeletionTimestamp.IsZero() {
+		// If drpolicy is deleted then return
+		// error to fail drpc reconciliation
+		return ctrl.Result{}, fmt.Errorf("drPolicy '%s' referred by the DRPC is deleted, DRPC reconciliation would fail",
+			drpc.Spec.DRPolicyRef.Name)
+	}
```

Comment:
@BenamarMk 
I have removed it from `createDRPCInstance` and added it here. So, do we update `DRPC` status here on failure. ?